include/arch/arm/arch/64/mode/object/structures.bf

```diff
@@ -44,43 +44,15 @@ block page_table_cap {
     padding                          20
 }
 
--- Third-level page table (page directory)
-block page_directory_cap {
-    field capPDMappedASID            16
-    field_high capPDBasePtr          48
-
-    field capType                    5
-    padding                          10
-    field capPDIsMapped              1
-    field_high capPDMappedAddress    19
-    padding                          29
-}
-
--- Second-level page table (page upper directory)
-block page_upper_directory_cap {
-    field capPUDMappedASID           16
-    field_high capPUDBasePtr         48
-
-    field capType                    5
-    field capPUDIsMapped             1
-    field_high capPUDMappedAddress   10
-#if defined (CONFIG_ARM_SMMU)  && defined (AARCH64_VSPACE_S2_START_L1)
-    field capPUDMappedCB             8
-    padding                          40
-#else 
-    padding                          48
-#endif 
-}
-
--- First-level page table (page global directory)
-block page_global_directory_cap {
-    field capPGDMappedASID           16
-    field_high capPGDBasePtr         48
+-- First-level page table (vspace_root)
```

Comment:
it looks like that the Arm arch manual uses level 0, 1, 2, 3 to refer the page tables. Should we follow the same pattern?

Reply:
I would slightly prefer the names, because the numbers can be read either way around (0 being bottom or 0 being top), and you have to know which way around the Arm manual uses them to read the code. With the names that possibility of confusion is gone. You do need to know which name maps to which level, so the gain is not much, but I'm still slightly in favour of that.